### PR TITLE
Set device in device dmatrix.

### PR DESCRIPTION
**Pull request by trivialfis**
Running pytest on all GPU tests on my machine resulted into a segfault.

### CODE REVIEW

**Comment by hcho3** [comment]
@trivialfis Does your machine have 2 GPUs?

**Comment by trivialfis** [comment]
@hcho3 Yup.

File: src/data/data.cc [hunk 1]
```diff
@@ -350,30 +350,44 @@ void MetaInfo::Validate() const {
         << "Invalid group structure.  Number of rows obtained from groups "
            "doesn't equal to actual number of rows given by data.";
   }
+  auto check_device = [device](HostDeviceVector<float> const &v) {
```
Comment (hcho3) [inline src/data/data.cc:353]: Should this be a proper function?

Comment (trivialfis) [reply src/data/data.cc:353]: @hcho3 I don't have a preference.  Maybe a member function of `HostDeviceVector`?  Or a general function in `device_helpers.cuh`?

Comment (hcho3) [reply src/data/data.cc:353]: It can go in `device_helpers.cuh`.

Comment (trivialfis) [reply src/data/data.cc:353]: @hcho3 No, my bad.  data.cc can not include a `cuh` file.

Comment (hcho3) [reply src/data/data.cc:353]: Okay, I suppose we can merge this PR as it is.

**Comment by hcho3** [comment]
@trivialfis This should be picked into `release_1.1.0` then?

**Comment by trivialfis** [comment]
@hcho3 Done.

**Pull request merged**